src/components/EditRecipe.tsx

```diff
@@ -0,0 +1,664 @@
+                  <InputGroup w="50%">
+                    <Input
+                      size="sm"
+                      value={recipe.recipeServings || ""}
+                      onChange={e => {
+                        setRecipe({
+                          ...recipe,
+                          recipeServings: Number(e.target.value)
+                        });
+                      }}
+                    />
+                    <InputRightElement marginRight="5">
+                      <Text fontSize="12">persons</Text>
+                    </InputRightElement>
+                  </InputGroup>
+                </FormControl>
+              </Flex>
+              <Box>
+                <Flex marginY="5">
+                  <Text marginRight="2">
+                    <b>Ingredients</b>
+                  </Text>
+                  <IconButton
+                    size="sm"
+                    variant="outline"
+                    aria-label="add ingredient"
+                    icon={<AddIcon />}
+                    title="add ingredient"
+                    onClick={() => handleInputAdd("ingredients")}
+                  />
+                </Flex>
+                <Flex justifyContent="center">
+                  <Text w="50%">
+                    <i>ingredient</i>
+                  </Text>
+                  <Text w="22%">
+                    <i>amount</i>
+                  </Text>
+                  <Text w="28%">
+                    <i>unit</i>
+                  </Text>
+                </Flex>
+                {ingredients.map((ingredient, index) => (
+                  <Flex key={index}>
+                    <FormControl marginRight="3">
+                      <FormLabel></FormLabel>
+                      <Input
+                        size="sm"
+                        name="ingredientName"
+                        type="text"
+                        value={ingredient.ingredientName}
+                        onChange={e => {
+                          const newIngredients = [...ingredients];
+                          newIngredients[index].ingredientName = e.target.value;
+                          setIngredients(newIngredients);
+                          setRecipe({
+                            ...recipe,
+                            recipeIngredients: newIngredients
+                          });
+                        }}
+                      />
+                    </FormControl>
+                    <FormControl w="40%" marginRight="3">
+                      <FormLabel></FormLabel>
+                      <Input
+                        size="sm"
+                        name="ingredientAmount"
+                        type="number"
+                        value={ingredient.ingredientAmount || ""}
+                        onChange={e => {
+                          const newIngredients = [...ingredients];
+                          newIngredients[index].ingredientAmount = Number(
```

Comment:
Probably a check should be added to not allow empty/zero value for ingredient amount. As of now it is possible to leave field empty and save recipe with 0 amount of an ingredient.

Reply:
I prohibit the oportunity to enter numbers less than 0 and handle the form rendering for numbers = 0. We discussed and desided to leave the oportinity fo user to enter just ingredients without amount